### PR TITLE
329 go native or go home

### LINKED ISSUE #329: [SPIKE] gitea [orgs/apps] -> kube [namespaces/deployments]
While we are trying to give less responsibilities to Gitea (See: #308, #328 etc), it makes sense to stop using gitea to list orgs and apps.
We can use kubernetes namespaces to group applications (as in "orgs") and we could list deployments in namespaces to find the applications in an org. This is more kubernetes native approach and also removes that responsibility from Gitea.

Unblocks: #103

### PR DESCRIPTION
**Pull request by jimmykarily**
Fixes #329

### CODE REVIEW

**Comment by jimmykarily** [comment]
Still missing the cleanup of the org/namespaces when doing `epinio uninstall`

**Comment by jimmykarily** [comment]
By moving the staging to a different namespace, this PR broke the staging logs of `epinio push --verbosity 1`. This is because of the way we tail for logs here: https://github.com/epinio/epinio/blob/main/internal/cli/clients/client.go#L1347

(changed here: https://github.com/epinio/epinio/pull/337/files#diff-9175d7addc590890bbee03a3eaaf4fedbee6f21a2f87d5993d3b17296283382dR1468-R1469)

We need to find a way to match the staging pods on the tekton-staging namespace too.

**Review by manno — APPROVED** [review]
LGTM

(maybe squash some commits and write an abstract?)

File: internal/cli/clients/client.go [hunk 1]
```diff
@@ -1274,15 +1287,124 @@ spec:
...
+  `)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse service template for app")
+	}
+
+	serviceFile, err := os.Create(filePath)
+	if err != nil {
+		return errors.Wrap(err, "failed to create file for application Service definition")
+	}
+	defer func() { err = serviceFile.Close() }()
```
Comment (manno) [inline internal/cli/clients/client.go:1335]: While this does make the linter happy, it does not check the err. I prefer 
```suggestion
	defer func() { _ = serviceFile.Close() }()
```

Or we disable the errchk linter?

Comment (jimmykarily) [reply internal/cli/clients/client.go]: done

**Comment by jimmykarily** [comment]
I squashed most commits on this one: https://github.com/epinio/epinio/pull/337/commits/bb2140351b7774d8499fcd5483c64c0f366993e2 and added an abstract in the commit message

**Comment by jimmykarily** [comment]
I have a fix for the logs on another branch. I will open a PR as soon as we merge this one.

**Comment by jimmykarily** [comment]
PR that fixes logs: https://github.com/epinio/epinio/pull/339

**Pull request merged**